source/opt/function.h

```diff
@@ -110,6 +110,10 @@ class Function {
                    bool run_on_debug_line_insts = false);
   void ForEachInst(const std::function<void(const Instruction*)>& f,
                    bool run_on_debug_line_insts = false) const;
+  bool WhileEachInst(const std::function<bool(Instruction*)>& f,
```

Comment:
It would be preferable to reimplement ForEachInst in terms of WhileEachInst now to avoid duplication of knowing the right traversal order.